src/ucp/core/ucp_worker.c

```diff
@@ -2711,7 +2711,11 @@ void ucp_worker_keepalive_remove_ep(ucp_ep_h ep)
         return;
     }
 
-    ucs_assert(!ucs_list_is_empty(&worker->all_eps));
+    /* MEM_TYPE EP's are not added to all_eps list and
+     * it may be empty, then just return from function */
+    if (ucs_list_is_empty(&worker->all_eps)) {
+        return;
+    }
```

Comment:
does it mean we call ucp_worker_keepalive_remove_ep() for memtype eps?

Reply:
yes - ```ucp_worker_keepalive_remove_ep``` is called from ```ucp_ep_delete``` which is general for all EPs

Reply:
so, for memtype ep, add is not called, but remove is called?
then, need to avoid calling remove - for symmetry

Reply:
there is no direct way to know if EP is mem_type...
add new flag?

Reply:
1. assert that ucp_ep_config(ep)->key.err_mode == UCP_ERR_HANDLING_MODE_NONE on memtype ep
2. skip remote, if ucp_ep_config(ep)->key.err_mode != UCP_ERR_HANDLING_MODE_NONE

Reply:
> 1. ucp_ep_config(ep)->key.err_mode == UCP_ERR_HANDLING_MODE_NONE

done

> 2\. skip remote, if ucp_ep_config(ep)->key.err_mode != UCP_ERR_HANDLING_MODE_NONE

will not work because keepalive iterator may stop on EP with NONE error mode

Reply:
maybe add UCP_EP_FLAG_TEMPORARY to memtype_ep?
and rename it to UCP_EP_FLAG_INTERNAL

Reply:
done

Reply:
fix for removed assert is reverted